### PR TITLE
chore: add eslint, prettier and commitlint

### LINKED ISSUE #12: Code Style, Formatting and DevOps
- [ ] Add linter
- [ ] Add prettier
- [ ] Add commitlint
- [ ] Add test cases
- [ ] Add CI/CD build pipeline
- [ ] Design guidelines
- [ ] Security
- [ ] Code of conduct

### PR DESCRIPTION
**Pull request by uwe**
Fixes #12 (partially)

Do you want to enforce a specific coding style?

PS: `eslint` needs an even node version (12, 14 or 16). `package.json` suggests that you are using 13.13. Any objections against an update to 14 or 16?

### CODE REVIEW

File: .eslintrc.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  project: "tsconfig.json"
+plugins:
+  - "@typescript-eslint"
+root: true
+rules:
+  "@typescript-eslint/no-floating-promises":
+    - error
+    - ignoreIIFE: true
+      ignoreVoid: true
+  "@typescript-eslint/no-inferrable-types": "off"
```
Comment (NouDaimon) [inline .eslintrc.yaml:17]: Perhaps we should set no-inferrable-types to "true"

**Review by NouDaimon — APPROVED** [review]

File: .tool-versions [hunk 2]
```diff
@@ -0,0 +1 @@
+nodejs 16.13.0
```
Comment (dhruvinparikh) [inline .tool-versions:1]: I think , this is not required.

**Review by dhruvinparikh — CHANGES_REQUESTED** [review]
`yarn-berry.cjs` is not required

File: .tool-versions (hunk 2 above)
Comment (uwe) [reply .tool-versions:1]: Do you use any version switchers for NodeJS? Is every developer using the same version of node?

I can remove that file (and also `yarn-berry.cjs`), but I will wait what the discussion brings.

Comment (NouDaimon) [reply .tool-versions:1]: I personally use NVM if this is what you mean Uwe? Most projects are built using a specified node version (please correct me if I am mistaken @dhruvinparikh ).

Comment (dhruvinparikh) [reply .tool-versions:1]: I do use nvm for switching

**Review by dhruvinparikh — APPROVED** [review]

**Pull request merged**